tests/auxiliary/minicore.rs

```diff
@@ -101,10 +102,101 @@ macro_rules! concat {
         /* compiler built-in */
     };
 }
+
 #[rustc_builtin_macro]
 #[macro_export]
 macro_rules! stringify {
     ($($t:tt)*) => {
         /* compiler built-in */
     };
 }
+
+#[macro_export]
+macro_rules! partialEq {
+    ($($ty:ty),* $(,)?) => {
+        $(
+            impl PartialEq for $ty {
+                #[inline]
+                fn eq(&self, other: &$ty) -> bool {
+                    (*self) == (*other)
+                }
+                #[inline]
+                fn ne(&self, other: &$ty) -> bool {
+                    (*self) != (*other)
+                }
+            }
+        )*
+    }
+}
+
+partialEq!(
+    bool, char, isize, i8, i16, i32, i64, i128, usize, u8, u16, u32, u64, u128, f16, f32, f64, f128
+);
+
+#[lang = "panic"]
+//#[rustc_const_panic_str]
+//#[inline(never)]
+//#[cold]
+//#[track_caller]
+//#[rustc_nounwind]
+const fn panic(expr: &'static str) -> ! {
+    abort()
+}
+
+#[lang = "panic_fmt"]
+const fn panic_fmt(fmt: &str) -> ! {
+    abort()
+}
```

Comment:
Problem: this doesn't match the real `panic_fmt`'s signature:

https://github.com/rust-lang/rust/blob/b522e7c5ea8334fbccb9dd7f9b33759304743fd0/library/core/src/panicking.rs#L49-L55

```rs
#[inline(never)]
#[cold]
#[track_caller]
#[lang = "panic_fmt"]
#[rustc_do_not_const_check]
pub const fn panic_fmt(fmt: fmt::Arguments<'_>) -> ! { 
```

I'm somewhat concerned about panic signatures divering between real core and minicore...

cc @oli-obk do you know if this is a problem?

Reply:
Not really. None of the other flags are really relevant to the caller or the language

Reply:
I'm surprised we don't have a check that the signature is correct. Getting the wrong signature is likely to lead to an ICE or miscompile in the future. For example, codegen generates calls to these lang items assuming that their signature is correct.

Reply:
Also opened https://github.com/rust-lang/rust/issues/137531